CollectorUtils/UpdateGNSSSMetadata.py

```diff
@@ -0,0 +1,293 @@
+import argparse
+import sys
+import urllib
+import urllib.parse
+import urllib.request
+import json
+from arcgis.gis import GIS
+
+
+# Parse Command-line arguments
+def parseArguments():
+    global args_parser
+    parser = argparse.ArgumentParser(
+        usage='Add/Update GNSS metadate fields')
+    parser.add_argument('-u', '--username', required=True, type=str, help='Organization username')
+    parser.add_argument('-p', '--password', required=True, type=str, help='Organization password')
+    parser.add_argument('-url', '--url', required=True, type=str, help='Organization url')
+    parser.add_argument('-r', '--remove', default=False, type=bool,
+                        help='Set True if GNSS metadata fields need to be removed')
+    parser.add_argument('itemId', type=str, nargs="+", help='Search string')
+    args_parser = parser.parse_args()
+
+
+# Generate Token based on username and password
+def generateToken():
+    global token
```

Comment:
Should not use global variables. Instead, return the token and pass it in to other functions you need.

Or if you are using the Python API, you can do gis._con to get a connection with POST and GET methods available.